ddtrace/contrib/wsgi/_utils.py

```diff
@@ -80,6 +80,9 @@ def __call__(self, environ, start_response):
             # This can occur if a streaming response exits abruptly leading to a broken pipe.
             # Note: The wsgi.response span will still have the error information included.
             req_span._ignore_exception(generatorExit)
+            # modify request span before calling the wsgi_app. This ensures the request span modifier
+            # is always called even when the wsgi app raises an exception.
+            self.request_span_modifier(req_span, environ)
```

Comment:
The last two commits contains a bug fix and a refactor somewhat related to the main change. 

Bug fix: [f03f8641929ba0fbc030faa2591ea0a8c1c98244](https://github.com/DataDog/dd-trace-py/pull/3907/commits/f03f8641929ba0fbc030faa2591ea0a8c1c98244), we should always set the resource name of wsgi.request to an endpoint even when an exception is raised in the wsgi app. 

Refactor: [85b068671aa6b14352368f1039506de36557074a](https://github.com/DataDog/dd-trace-py/pull/3907/commits/85b068671aa6b14352368f1039506de36557074a), removes duplicate snapshots.

I cherry pick these commits and propose these changes in a follow up PR.